spyder/plugins/tests/test_ipythonconsole.py

```diff
@@ -857,5 +858,77 @@ def test_set_elapsed_time(ipyconsole, qtbot):
     client.timer.timeout.disconnect(client.show_time)
 
 
+@pytest.mark.slow
+@flaky(max_runs=3)
+def test_stderr_file_is_remove_one_kernel(ipyconsole, qtbot, monkeypatch):
+    """Test that consoles removes stderr when client is close."""
+    # Wait until the window is fully up
+    shell = ipyconsole.get_current_shellwidget()
+    client = ipyconsole.get_current_client()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+
+    # In a normal situation file should exists
+    monkeypatch.setattr(QMessageBox, 'question',
+                        classmethod(lambda *args: QMessageBox.Yes))
+    assert osp.exists(client.stderr_file)
+    ipyconsole.close_client(client=client)
+    assert not osp.exists(client.stderr_file)
+
+
+@pytest.mark.slow
+@flaky(max_runs=3)
+def test_stderr_file_is_remove_two_kernels(ipyconsole, qtbot, monkeypatch):
+    """Test that console removes stderr when client and related clients
+    are close."""
+    # Wait until the window is fully up
+    shell = ipyconsole.get_current_shellwidget()
+    client = ipyconsole.get_current_client()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+
+    # New client with the same kernel
+    ipyconsole._create_client_for_kernel(client.connection_file, None, None,
+                                         None)
+
+    assert len(ipyconsole.get_related_clients(client)) == 1
+    other_client = ipyconsole.get_related_clients(client)[0]
+    assert client.stderr_file == other_client.stderr_file
+
+    # In a normal situation file should exists
+    monkeypatch.setattr(QMessageBox, 'question',
+                        classmethod(lambda *args: QMessageBox.Yes))
+    assert osp.exists(client.stderr_file)
+    ipyconsole.close_client(client=client)
+    assert not osp.exists(client.stderr_file)
+
+
+@pytest.mark.slow
+@flaky(max_runs=3)
+def test_stderr_file_remains_two_kernels(ipyconsole, qtbot, monkeypatch):
+    """Test that console doesn't remove stderr when a related client is not
+    close."""
+    # Wait until the window is fully up
+    shell = ipyconsole.get_current_shellwidget()
+    client = ipyconsole.get_current_client()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+
+    # New client with the same kernel
+    ipyconsole._create_client_for_kernel(client.connection_file, None, None,
+                                         None)
+
+    assert len(ipyconsole.get_related_clients(client)) == 1
+    other_client = ipyconsole.get_related_clients(client)[0]
+    assert client.stderr_file == other_client.stderr_file
+
+    # In a normal situation file should exists
```

Comment:
`exists` -> `exist`

Reply:
I think that I need to check my English... :)

Reply:
No problem, these were very simple mistakes ;-)